### PR TITLE
grpc to support empty response.

### PR DESCRIPTION
**Pull request by qiwzhang**

### CODE REVIEW

File: source/common/grpc/rpc_channel_impl.cc [hunk 1]
```diff
@@ -54,7 +54,7 @@ void RpcChannelImpl::onSuccess(Http::MessagePtr&& http_response) {
 
     // A gRPC response contains a 5 byte header. Currently we only support unary responses so we
     // ignore the header. @see serializeBody().
-    if (!http_response->body() || !(http_response->body()->length() > 5)) {
+    if (!http_response->body() || !(http_response->body()->length() >= 5)) {
```
Comment (RomanDzhabarov) [inline source/common/grpc/rpc_channel_impl.cc]: nit: http_response->body()->length() < 5, seems to be more readable

File: test/common/grpc/rpc_channel_impl_test.cc [hunk 2]
```diff
@@ -212,6 +212,26 @@ TEST_F(GrpcRequestImplTest, ShortBodyInResponse) {
...
+  helloworld::HelloRequest request;
+  request.set_name("a name");
+  helloworld::HelloReply response;
+  EXPECT_CALL(grpc_callbacks_, onPreRequestCustomizeHeaders(_));
+  service_.SayHello(nullptr, &request, &response, nullptr);
+
+  Http::MessagePtr response_http_message(new Http::ResponseMessageImpl(
+      Http::HeaderMapPtr{new Http::TestHeaderMapImpl{{":status", "200"}}}));
+  helloworld::HelloReply inner_response;
+  response_http_message->body() = Common::serializeBody(inner_response);
```
Comment (htuch) [inline test/common/grpc/rpc_channel_impl_test.cc]: Can you add an `EXPECT` on the serialized body size being 5 bytes?

File: test/common/grpc/rpc_channel_impl_test.cc [hunk 3]
```diff
@@ -212,6 +212,26 @@ TEST_F(GrpcRequestImplTest, ShortBodyInResponse) {
...
+
+  helloworld::HelloRequest request;
+  request.set_name("a name");
+  helloworld::HelloReply response;
+  EXPECT_CALL(grpc_callbacks_, onPreRequestCustomizeHeaders(_));
+  service_.SayHello(nullptr, &request, &response, nullptr);
+
+  Http::MessagePtr response_http_message(new Http::ResponseMessageImpl(
+      Http::HeaderMapPtr{new Http::TestHeaderMapImpl{{":status", "200"}}}));
+  helloworld::HelloReply inner_response;
```
Comment (RomanDzhabarov) [inline test/common/grpc/rpc_channel_impl_test.cc]: nit: empty_response might be more clear

File: test/common/grpc/rpc_channel_impl_test.cc (hunk 2 above)
Comment (qiwzhang) [inline test/common/grpc/rpc_channel_impl_test.cc]: Done

File: test/common/grpc/rpc_channel_impl_test.cc (hunk 3 above)
Comment (qiwzhang) [inline test/common/grpc/rpc_channel_impl_test.cc]: Done

File: source/common/grpc/rpc_channel_impl.cc (hunk 1 above)
Comment (qiwzhang) [inline source/common/grpc/rpc_channel_impl.cc]: Done

**Pull request merged**